version: "3.9"

name: diploma-local
services:
  django:
    build:
      context: .
      dockerfile: docker/django/local/Dockerfile
    volumes:
      - .:/diploma/app
    ports:
      - "8000:8000"
    env_file:
      - .env.local
    command: ["/scripts/application.sh"]
    depends_on:
      database:
        condition: service_healthy
      rabbitmq:
        condition: service_started
      redis:
        condition: service_started
  database:
    image: postgres:15
    volumes:
      - postgres_data:/var/lib/postgresql/data/
      - ./db_backup:/db_backup
      - ./db_backup/dump.sql:/docker-entrypoint-initdb.d/dumpfile.sql
    env_file:
      - .env.local
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}']
      interval: 2s
      timeout: 5s
      retries: 5

  celery_worker:
    build:
      context: .
      dockerfile: docker/django/local/Dockerfile
    volumes:
      - .:/diploma/app
    env_file:
      - .env.local
    command: ["/scripts/celery.sh"]

  rabbitmq:
    image: rabbitmq:3.12

  redis:
    image: redis:7.2

volumes:
  postgres_data:
